graphql/util.lua

```diff
@@ -38,6 +47,20 @@ function util.coerceValue(node, schemaType, variables)
     return variables[node.name.value]
   end
 
+  if schemaType.__type == 'Union' then
+    local matched, result
+    for k, v in ipairs(schemaType.types) do
+        matched, result = pcall(util.coerceValue, node, v, variables)
+        if matched then
+          return result
+        end
+    end
+
+    local valid_types = util.map(schemaType.types, function (t) return t.name end)
+    --TODO! how to better report this error?
+    error('Could not coerce "' .. tostring(node.kind) .. '" to one of union types ['..table.concat(valid_types, ",")..']')
+  end
```

Comment:
I think you are referring to [this line](https://github.com/bjornbytes/graphql-lua/blob/master/graphql/execute.lua#L186) when you say unions should be resolved and it's true for ast nodes, but coerceValue is used when working with arguments, for example [here](https://github.com/bjornbytes/graphql-lua/blob/master/graphql/execute.lua#L211) and also in the validation (rules) section. So at least as it is now, "argument" nodes do not get their "types resolved", hence this code.
But you might be right, maybe the types should be resolved somewhere up the chain, i'll look into it if there is a more appropriate place